pkg/plugin/clientmgmt/manager.go

```diff
@@ -266,3 +255,14 @@ func (m *manager) GetDeleteItemAction(name string) (velero.DeleteItemAction, err
 	r := newRestartableDeleteItemAction(name, restartableProcess)
 	return r, nil
 }
+
+// sanitizeName adds "velero.io" to legacy plugins that weren't namespaced.
+func sanitizeName(name string) string {
+	// Backwards compatibility with non-namespaced Velero plugins, following principle of least surprise
+	// since DeleteItemActions were not bundled with Velero when plugins were non-namespaced.
+	if !strings.Contains(name, "/") {
```

Comment:
If we wanted to be really strict, we know we support 4 plugins, and we could make this function validate against those 4 so that people _know_ this is off-limits.

I wonder if Kubernetes does this for Kubernetes namespaced labels.

Reply:
Yeah, I had gone that route, then backed away.

It's not only 4, it's all of these (I didn't prune the dups):

```
velero.io/crd-remap-version          BackupItemAction
velero.io/pod                        BackupItemAction
velero.io/pv                         BackupItemAction
velero.io/service-account            BackupItemAction
velero.io/aws                        ObjectStore
velero.io/add-pv-from-pvc            RestoreItemAction
velero.io/add-pvc-from-pod           RestoreItemAction
velero.io/change-pvc-node-selector   RestoreItemAction
velero.io/change-storage-class       RestoreItemAction
velero.io/cluster-role-bindings      RestoreItemAction
velero.io/crd-preserve-fields        RestoreItemAction
velero.io/init-restore-hook          RestoreItemAction
velero.io/job                        RestoreItemAction
velero.io/pod                        RestoreItemAction
velero.io/restic                     RestoreItemAction
velero.io/role-bindings              RestoreItemAction
velero.io/service                    RestoreItemAction
velero.io/service-account            RestoreItemAction
velero.io/aws                        VolumeSnapshotter
```

If we do this, plugins like tardigate will have to be updated to not user the `velero.io` namespace (granted, it wasn't documented so it probably seemed like a very logical thing to do).

Reply:
D'oh, you're right on the internal ones!

Honestly, I'm not too concerned about breaking external folks using our domain in their plugin, because we _will_ end up doing that at some point. We'll have to drop the `velero.io` usage someday, as documented in our code.

Reply:
I'll mark this as resolved. We don't have to get super strict, and I don't want to enumerate everything, either. :)

Reply:
Ok so should I do it? I literally had this code working and deleted it (argh duh darn it I didn't save it). I think it's a good house keeping code, glad to redo it.

Reply:
I responded to probably a stale version of the thread, hadn't seen your last msg. Ok, we keep it as is.

Reply:
Yeah, keep the existing check, gathered into a single function. That's 👍  for me.